### PR TITLE
Return `undefined` if the found `node_modules` directory is not writable

### LINKED ISSUE #10: Return `null` if the found `node_modules` directory is not writable
When building node modules with Nix/node2nix `node_modules` is pre-built and mounted read-only, resulting in `node_modules/.cache` not being writable. A nice and easy possibility to change the directory would be to allow overriding it with an ENV var (`NODE_CACHE_DIR`?)

### PR DESCRIPTION
**Pull request by Richienb**
Closes #10

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -17,6 +18,10 @@ module.exports = (options = {}) => {
 	directory = pkgDir.sync(directory);
 
 	if (directory) {
+		if (!isWritable.sync(path.join(directory, 'node_modules'))) {
```
Comment (sindresorhus) [inline index.js]: We don't need a module for this. Just add a helper method:

```js
const isWritable = path => {
	try {
		fs.accessSync(path, fs.constants.W_OK);
		return true
	} catch (_) {
		return false
	}
};
```

File: index.js [hunk 2]
```diff
@@ -17,6 +18,10 @@ module.exports = (options = {}) => {
...
+			return null;
```
Comment (sindresorhus) [inline index.js]:
```suggestion
			return undefined;
```

**Comment by sindresorhus** [comment]
The behavior should be documented.

File: index.js (hunk 1 above)
Comment (Richienb) [reply index.js]: Refactored.

**Comment by Richienb** [comment]
@sindresorhus

**Comment by Richienb** [comment]
@sindresorhus This PR is ready for a re-review.

**Pull request merged**